tests/test_nonbonded_interaction_group.py

```diff
@@ -0,0 +1,144 @@
+
+
+@pytest.fixture
+def example_system():
+    pdb_path = "tests/data/5dfr_solv_equil.pdb"
+    host_pdb = app.PDBFile(pdb_path)
+    ff = app.ForceField("amber99sbildn.xml", "tip3p.xml")
+    return (
+        ff.createSystem(host_pdb.topology, nonbondedMethod=app.NoCutoff, constraints=None, rigidWater=False),
+        host_pdb.positions,
+        host_pdb.topology.getPeriodicBoxVectors(),
+    )
+
+
+@pytest.fixture
+def example_nonbonded_potential(example_system):
+    host_system, _, _ = example_system
+    host_fns, _ = openmm_deserializer.deserialize_system(host_system, cutoff=1.0)
+
+    nonbonded_fn = None
+    for f in host_fns:
+        if isinstance(f, potentials.Nonbonded):
+            nonbonded_fn = f
+
+    return nonbonded_fn
+
+
+@pytest.fixture
+def example_coords(example_system):
+    _, host_coords, _ = example_system
+    return np.array([[to_md_units(x), to_md_units(y), to_md_units(z)] for x, y, z in host_coords])
+
+
+@pytest.fixture
+def example_box(example_system):
+    _, _, box = example_system
+    return np.asarray(box / box.unit)
+
+
+@pytest.mark.parametrize("beta", [2.0])
+@pytest.mark.parametrize("cutoff", [1.1])
+@pytest.mark.parametrize("precision,rtol,atol", [(np.float64, 1e-8, 1e-8), (np.float32, 1e-4, 5e-4)])
+@pytest.mark.parametrize("num_row_atoms", [1, 15])
+@pytest.mark.parametrize("num_atoms", [33, 231, 1050])
+def test_nonbonded_interaction_group_correctness(
+    num_atoms,
+    num_row_atoms,
+    precision,
+    rtol,
+    atol,
+    cutoff,
+    beta,
+    example_nonbonded_potential,
+    example_coords,
+    example_box,
+    rng,
+):
+
+    coords = example_coords[:num_atoms]
+    params = example_nonbonded_potential.params[:num_atoms, :]
+    lambda_offset_idxs = np.zeros(num_atoms, dtype=np.int32)
+
+    def make_reference_nonbonded(lambda_plane_idxs):
+        return prepare_reference_nonbonded(
+            params=params,
+            exclusion_idxs=np.array([], dtype=np.int32),
+            scales=np.zeros((0, 2), dtype=np.float64),
+            lambda_plane_idxs=lambda_plane_idxs,
+            lambda_offset_idxs=lambda_offset_idxs,
+            beta=beta,
+            cutoff=cutoff,
+        )
+
+    ref_allpairs = make_reference_nonbonded(np.zeros(num_atoms, dtype=np.int32))
+
+    row_atom_idxs = rng.choice(num_atoms, size=num_row_atoms, replace=False).astype(np.int32)
+    lambda_plane_idxs = np.zeros(num_atoms, dtype=np.int32)
+    lambda_plane_idxs[row_atom_idxs] = 1
+
+    ref_allpairs_minus_ixngroups = make_reference_nonbonded(lambda_plane_idxs)
+
+    def ref_ixngroups(*args):
+        return ref_allpairs(*args) - ref_allpairs_minus_ixngroups(*args)
+
+    test_ixngroups = NonbondedInteractionGroup(
+        row_atom_idxs,
+        np.zeros(num_atoms, dtype=np.int32),  # lambda plane indices
+        lambda_offset_idxs,
+        beta,
+        cutoff,
+    )
+
+    GradientTest().compare_forces(
+        coords,
+        params,
+        example_box,
+        lamb=0.1,
```

Comment:
hmm, since we're testing with a non-interpolated potential and `lambda_offset_idxs` are all zero, would the varying the lambda value make a difference here? It does seem like we should add a test covering the interpolated case, similar to what's in `test_parameter_interpolation.py`?

Reply:
It shouldn't make a difference, but no little cost to testing it

Reply:
We already do here: https://github.com/proteneer/timemachine/blob/master/tests/test_nonbonded.py#L471

Reply:
ah, I see, I had missed that case (we don't do it in `test_correctness`, which is the test I had originally adapted from)

Reply:
Added multiple lambda values to both the correctness and consistency tests in 
dd543ea.